### PR TITLE
KYP-762: Added fallback for option ID if SKU is not present

### PR DESCRIPTION
**Pull request by nedick007**
For legacy reasons m1 projects are sending product option sku instead of id, this modification should handle the migration from m1 to m2 module.

### CODE REVIEW

File: Helper/DeletedProductSerializer.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ public function serialize($deletedProductOrders)
                     
                     if ($parentProduct) {
                         $productOptions[] = [
-                            'id'       => $itemId,
+                            'id'       => ($itemSku) ? $itemSku : $itemId,
```
Comment (evgeni-spasov-metrilo) [inline Helper/DeletedProductSerializer.php]: Those are some magic parentheses?

File: Model/Events/RemoveFromCart.php [hunk 2]
```diff
@@ -10,6 +10,24 @@ public function __construct(
...
+            if ($itemSku) {
+                $itemId = $itemSku;
+            } else {
+                $itemId = $this->event->getQuoteItem()->getChildren()[0]->getProductId();
+            }
+        } else {
+            $itemId = $this->event->getQuoteItem()->getProductId();
+        }
+        
+        return $itemId;
```
Comment (evgeni-spasov-metrilo) [inline Model/Events/RemoveFromCart.php]: See above.

File: Model/Events/AddToCart.php [hunk 3]
```diff
@@ -10,6 +10,24 @@ public function __construct(
...
+            if ($itemSku) {
+                $itemId = $itemSku;
+            } else {
+                $itemId = $this->event->getQuoteItem()->getChildren()[0]->getProductId();
+            }
+        } else {
+            $itemId = $this->event->getQuoteItem()->getProductId();
+        }
+        
+        return $itemId;
```
Comment (evgeni-spasov-metrilo) [inline Model/Events/AddToCart.php]:
```php
    private function getItemIdentifier() {
        $item = $this->event->getQuoteItem();
        $itemOptions = $item->getChildren();

        if(!$itemOptions) {
            return $item->getProductId(); 
        end
        
        $itemSku = $itemOptions[0]->getSku();
        
        if ($itemSku) {
            return $itemSku;
        } else {
            return $itemOptions[0]->getProductId();
        }
    }
```
OR
```php
    private function getItemIdentifier() {
        $item = $this->event->getQuoteItem();
        $itemOptions = $item->getChildren();
        
        if ($itemOptions) {
            $itemSku = $itemOptions[0]->getSku();
            
            if ($itemSku) {
                return $itemSku;
            } else {
                return $itemOptions[0]->getProductId();
            }
        }

        return $item->getProductId();
    }
```

File: Helper/DeletedProductSerializer.php (hunk 1 above)
Comment (nedick007) [reply Helper/DeletedProductSerializer.php]: Parentheses are used only if there is expression, removing them.

File: Model/Events/RemoveFromCart.php (hunk 2 above)
Comment (nedick007) [reply Model/Events/RemoveFromCart.php]: Refactored.

File: Model/Events/AddToCart.php (hunk 3 above)
Comment (nedick007) [reply Model/Events/AddToCart.php]: Refactored.

**Comment by nedick007** [comment]
@evgeni-spasov-metrilo ready for review.

**Comment by nedick007** [comment]
@evgeni-spasov-metrilo ready for review

**Pull request merged**